src/adapter/src/catalog.rs

```diff
@@ -2485,6 +2485,40 @@ impl<S: Append> Catalog<S> {
         if let Some(system_parameter_frontend) = config.system_parameter_frontend {
             if !catalog.state.system_config().config_has_synced_once() {
                 tracing::info!("parameter sync on boot: start sync");
+
+                // We intentionally block initial startup, potentially forever,
+                // on initializing LaunchDarkly. This may seem scary, but the
+                // alternative is even scarier. Over time, we expect that the
+                // compiled-in default values for the system parameters will
+                // drift substantially from the defaults configured in
+                // LaunchDarkly, to the point that starting an environment
+                // without loading the latest values from LaunchDarkly will
+                // result in running an untested configuration.
+                //
+                // Note this only applies during initial startup. Restarting
+                // after we've synced once doesn't block on LaunchDarkly, as it
+                // seems reasonable to assume that the last-synced configuration
+                // was valid enough.
+                //
+                // This philosophy appears to provide a good balance between
+                // not running untested configurations in production while also
+                // not making LaunchDarkly a "tier 1" dependency for existing
+                // environments.
+                //
+                // If it is absolutely necessary to bring up a new environment
+                // while LaunchDarkly is down, the following manual mitigation
+                // can be performed:
+                //
+                //    1. Edit the environmentd startup parameters to omit
```

Comment:
Or we can pass `config_has_synced_once = true` as a bootstrap parameter env var once?

Reply:
This needs to be checked, as the bootstrap overrides are not persisted at the moment. This might still work, though, but not 100% confident at the moment.

Reply:
Gotcha. I'll leave as is for the moment, then. Thanks!